core/commands/cmdenv/env.go

```diff
@@ -70,3 +71,28 @@ func GetConfigRoot(env cmds.Environment) (string, error) {
 
 	return ctx.ConfigRoot, nil
 }
+
+// EscNonPrint converts non-printable characters and backslash into Go
+// escape sequences, if the given string contains any.
```

Comment:
WDYT about adding a comment that we're trying to mimic the escape quoting style from https://www.gnu.org/software/coreutils/manual/html_node/Formatting-the-file-names.html?

Reply:
The link says essentially what the comment here says, so I don't think it adds much, other than it hints that we are doing this for the same reason.  Also, I fear that people will misread that and think it means identical content.

Added brief description of purpose.

Reply:
Fair enough, if/when we add other escaping options we can add more descriptions